catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorEditTextGenericTests.java

```diff
@@ -0,0 +1,120 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.formulaeditor;
+
+import android.support.annotation.StringRes;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.ChangeSizeByNBrick;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+
+import java.util.Arrays;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.uiespresso.formulaeditor.utils.FormulaEditorWrapper.onFormulaEditor;
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResources;
+
+@Category({Cat.AppUi.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
+public class FormulaEditorEditTextGenericTests {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("FormulaEditorEditTextTest");
+		script.addBrick(new ChangeSizeByNBrick());
+		baseActivityTestRule.launchActivity(null);
+		onView(withId(R.id.brick_change_size_by_edit_text))
+				.perform(click());
+	}
+
+	@Parameterized.Parameters(name = "{2}" + "-Test")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{R.id.formula_editor_keyboard_0, "0 ", "button_0"},
+				{R.id.formula_editor_keyboard_1, "1 ", "button_1"},
+				{R.id.formula_editor_keyboard_2, "2 ", "button_2"},
+				{R.id.formula_editor_keyboard_3, "3 ", "button_3"},
+				{R.id.formula_editor_keyboard_4, "4 ", "button_4"},
+				{R.id.formula_editor_keyboard_5, "5 ", "button_5"},
+				{R.id.formula_editor_keyboard_6, "6 ", "button_6"},
+				{R.id.formula_editor_keyboard_7, "7 ", "button_7"},
+				{R.id.formula_editor_keyboard_8, "8 ", "button_8"},
+				{R.id.formula_editor_keyboard_9, "9 ", "button_9"},
+				{R.id.formula_editor_keyboard_decimal_mark, "0" + getResources().getString(R.string
```

Comment:
theres also getResourcesString in uitestutils, which to further increase readability you could proxy through a new method in this test to eg. getString().
if you add the " " in the test where you check them it would clean this up significantly...

Reply:
I don't really get what you mean by "proxying through a new method" and using getResourcesString doesn't improve anything and IMO it makes it even worse since android studio already has the option to only show the text "." instead of getResources().getString(...) while it shows getResourcesString(".") if I use the function. Please see the screenshot below if this explanation isn't clear :S
![Difference between the two scenarios](https://imgur.com/a/SCxk7)

For me, it doesn't make sense to use getResourcesString() over the getResources().getString() since it makes it harder to read because android studio doesn't know that function and hence it can't view it simplified. However, if you really want it this way I can, of course, change it to this I'm just unsure if I understand correctly what you mean...